chapters/os_tooling.md

```diff
@@ -0,0 +1,84 @@
+# Offline Compilation of OpenCL Kernel Sources
+
+Aside from online compilation during application execution, OpenCL kernel sources can be compiled offline into binaries that can be loaded into the drivers using special API calls (e.g. `clCreateProgramWithBinary` or `clCreateProgramWithIL`).
+
+This section describes available open source tools for offline compilation of OpenCL kernels.
+
+## Open Source Tools
+
+* [clang](https://clang.llvm.org/) is a compiler front-end for the C/C++ family of languages including OpenCL C and C++ for OpenCL that can produce executable binaries (e.g. AMDGPU), or portable binaries (e.g. SPIR). It is part of [the LLVM compiler infrastructure project](https://llvm.org/), and there is information regarding [OpenCL kernel language support and standard headers](https://clang.llvm.org/docs/UsersManual.html#opencl-features).
```

Comment:
I wonder if it makes more sense to mention LLVM IR here vs. SPIR.  Note that the diagram below says "LLVM IR".

Yes, there is not much of a difference between the two in practice...

Reply:
Good point. Actually, I wonder if it's better to say "SPIR flavor of LLVM IR" here because technically it is no longer a conformant SPIR.

Reply:
FYI I have added this clarification later where individual commands are explained. Otherwise it might not be clear enough here.